### PR TITLE
feat: Support importing existing VPC into rift_compute module

### PR DESCRIPTION
**Pull request by eshiferax**
1. Updating new `modules/` to take `aws` provider as _input_ rather than defining in-module. This is necessary for the `data` resources (like `data "aws_availability_zones" "available"`) to use the expected role/provider. 

2. Adding 3 new inputs to `rift_compute` module (and by extension, `dataplane_rift`, `dataplane_rift_with_emr`, and `standalone_rift` modules)
     * `existing_vpc` (object)
          * `vpc_id`
          * `private_subnet_ids`

     * `existing_rift_compute_security_group_id`

       These are optional -- if set, a VPC will _not_ be created for Rift, and existing resources will be used.

Tested with:
   1. Existing Rift compute cluster. Set git reference to this branch (`existing-vpc-rift`) and did _not_ provide values for the new^ variables. This resulted in `No changes. Your infrastructure matches the configuration.`
   2. New cluster. Created VPC in separate module, provided as input to [dataplane_rift](https://github.com/tecton-ai/tecton-terraform-setup/tree/master/modules/dataplane_rift) module. All resources, roles/policies, etc, were created as expected.
      * Also tested the validation conditions for the `existing_vpc` object (empty values for either property)

### CODE REVIEW

File: modules/standalone_rift/variables.tf [hunk 1]
```diff
@@ -28,6 +28,25 @@ variable "tecton_control_plane_role_name" {
   type        = string
 }
 
+variable "existing_vpc_id" {
```
[the remark this answers is not in the document]
Comment (eshiferax) [reply modules/standalone_rift/variables.tf]: @zhoujoetan  Yes, the 'existing' VPC ID and Private Subnet ID variables are coupled (can't be enabled exclusively). I updated them to be grouped into the same `existing_vpc` object, and added validation as you describe. Tested with the whole thing being null (pass), one or the other being null/empty (fail), and both being set (pass). 

Left the existing security-group ID as a separate variable because it can be either included or not regardless of the other 2.

**Review by zhoujoetan — APPROVED** [review]
